tests/core/conversations_worker/test_event_publisher.py

```
@@ -281,17 +281,68 @@ def test_publisher_batches_intermediate_events():
+                StreamMessage(event=StreamEvents.TOKEN_COUNT, data={"k": "post-llm"}),
+                StreamMessage(event=StreamEvents.START_TOOL, data={"tool_name": "t1"}),
+                StreamMessage(event=StreamEvents.TOOL_RESULT, data={"tool_call_id": "1"}),
+                StreamMessage(event=StreamEvents.TOOL_RESULT, data={"tool_call_id": "2"}),
+                StreamMessage(event=StreamEvents.TOKEN_COUNT, data={"k": "post-tools"}),
+                # Second LLM iteration: final answer
+                StreamMessage(event=StreamEvents.ANSWER_END, data={"content": "ok"}),
+            ]
+        )
+    )
+    # First flush:  AI_MESSAGE + TOKEN_COUNT       (triggered by TOKEN_COUNT #1)
+    # Second flush: START_TOOL + TOOL_RESULT × 2 + TOKEN_COUNT
```

Ruff (0.15.12)

[warning] Comment contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF003)